[webpack] Web Apps

orangejenny opened the pull request:
## Technical Description
This is not a whole lot of code, since web apps was already on requirejs. Mostly it's switching from the old `hqRequire` approach for circular dependencies to using `import` calls, plus a few config updates relating to external libraries that haven't been used in a webpack context yet.

https://dimagi.atlassian.net/browse/SAAS-16004

## Safety Assurance

### Safety story
This will go through a full regression test.

### Automated test coverage

A little bit.

### QA Plan

https://dimagi.atlassian.net/browse/QA-7384

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

orangejenny committed:
Updated base HTML templates and added commcarehq dependency

orangejenny committed:
Replaced hqRequire with import

orangejenny committed:
Fixed calendar picker

In requirejs, these dependencies were part of the requirejs config:
https://github.com/dimagi/commcare-hq/blob/e999662c9c2d78d90e5dfdbf1e2d5b895edf4c40/corehq/apps/hqwebapp/static/hqwebapp/js/bootstrap5/requirejs_config.js#L37-L42

orangejenny committed:
Removed $ from template inline js

orangejenny committed:
Added exports-loader config for mapbox

orangejenny committed:
Updated sentry usage

millerdev commented on corehq/apps/cloudcare/templates/cloudcare/preview_app_base.html:
```diff
@@ -51,13 +51,14 @@
   <script src="{% statici18n LANGUAGE_CODE %}"></script> {# DO NOT COMPRESS #}
 
   {% include "hqwebapp/partials/requirejs.html" %}
+  {% include "hqwebapp/partials/webpack.html" %}
```
Is it fine to be including both of these? Seems like it should be one or the other.

orangejenny replied:
It's not harmful, because all the logic in `requirejs.html` checks for a `requirejs_main` value, but yes it can be removed, I just missed it. ff214d8195e

MartinRiese reviewed (approved):
Looks good to me

orangejenny committed:
Removed include of requirejs.html

orangejenny committed:
Migratede cloudcare tests to webpack

orangejenny committed:
Merge branch 'master' into jls/webpack-web-apps

orangejenny commented:
I merged master to pull in [this PR](https://github.com/dimagi/commcare-hq/pull/35623), then updated cloudcare tests to run using webpack insetad of requirejs.

orangejenny committed:
Added eslint config for cloudcare that uses latest ECMA version

The repo-wide config uses 6, but that doesn't allow for "import" not at the top level,
which is a parse error that then obscures all other lint errors.

orangejenny committed:
Fixed lint errors

orangejenny committed:
Updated mapping code

orangejenny committed:
Updated maps dependencies (QA-7401)
